kanidmd/src/lib/be/idl_sqlite.rs

```diff
@@ -220,153 +212,122 @@ pub trait IdlSqliteTransaction {
+                .optional()
+                .map_err(sqlite_error)?;
 
-                let uuid = uuid_raw.as_ref().and_then(|u| Uuid::parse_str(u).ok());
-                trace!(%name, ?uuid, "Got uuid for index");
-                ltrace!(audit, "Got uuid for index name {} -> {:?}", name, uuid);
+            let uuid = uuid_raw.as_ref().and_then(|u| Uuid::parse_str(u).ok());
+            trace!(%name, ?uuid, "Got uuid for index");
 
-                Ok(uuid)
-            })
+            Ok(uuid)
         })
     }
 
     // ! TRACING INTEGRATED
-    fn uuid2spn(
-        &mut self,
-        audit: &mut AuditScope,
-        uuid: &Uuid,
-    ) -> Result<Option<Value>, OperationError> {
+    fn uuid2spn(&mut self, uuid: &Uuid) -> Result<Option<Value>, OperationError> {
         spanned!("be::idl_sqlite::uuid2spn", {
-            lperf_trace_segment!(audit, "be::idl_sqlite::uuid2spn", || {
-                let uuids = uuid.to_hyphenated_ref().to_string();
-                // The table exists - lets now get the actual index itself.
-                let mut stmt = self
-                    .get_conn()
-                    .prepare("SELECT spn FROM idx_uuid2spn WHERE uuid = :uuid")
-                    .map_err(sqlite_error)?;
-                let spn_raw: Option<Vec<u8>> = stmt
-                    .query_row(&[(":uuid", &uuids)], |row| row.get(0))
-                    // We don't mind if it doesn't exist
-                    .optional()
-                    .map_err(sqlite_error)?;
+            let uuids = uuid.to_hyphenated_ref().to_string();
+            // The table exists - lets now get the actual index itself.
+            let mut stmt = self
+                .get_conn()
+                .prepare("SELECT spn FROM idx_uuid2spn WHERE uuid = :uuid")
+                .map_err(sqlite_error)?;
+            let spn_raw: Option<Vec<u8>> = stmt
+                .query_row(&[(":uuid", &uuids)], |row| row.get(0))
+                // We don't mind if it doesn't exist
+                .optional()
+                .map_err(sqlite_error)?;
 
-                let spn: Option<Value> = match spn_raw {
-                    Some(d) => {
-                        let dbv = serde_cbor::from_slice(d.as_slice()).map_err(serde_cbor_error)?;
-                        let spn = Value::from_db_valuev1(dbv)
-                            .map_err(|_| OperationError::CorruptedIndex("uuid2spn".to_string()))?;
-                        Some(spn)
-                    }
-                    None => None,
-                };
+            let spn: Option<Value> = match spn_raw {
+                Some(d) => {
+                    let dbv = serde_cbor::from_slice(d.as_slice()).map_err(serde_cbor_error)?;
+                    let spn = Value::from_db_valuev1(dbv)
+                        .map_err(|_| OperationError::CorruptedIndex("uuid2spn".to_string()))?;
+                    Some(spn)
+                }
+                None => None,
+            };
```

Comment:
if-let!

Reply:
See above, logic change. This is old code mate!